go-controller/pkg/csrapprover/approver.go

```diff
@@ -78,51 +92,48 @@ var Predicate = predicate.Funcs{
 
 // NewController creates a new OVNKubeCSRController
 func NewController(client crclient.Client,
-	commonNamePrefixes string,
-	organization []string,
-	groups, userPrefixes sets.Set[string],
+	csrAcceptanceConditions []CSRAcceptanceCondition,
 	usages sets.Set[certificatesv1.KeyUsage],
 	maxDuration time.Duration,
 	recorder record.EventRecorder) *OVNKubeCSRController {
-	c := &OVNKubeCSRController{
-		name:               ControllerName,
-		client:             client,
-		commonNamePrefixes: commonNamePrefixes,
-		organization:       organization,
-		groups:             groups,
-		usages:             usages,
-		userPrefixes:       userPrefixes,
-		maxDuration:        maxDuration,
-		recorder:           recorder,
-	}
-	return c
-}
 
-func (c *OVNKubeCSRController) filterCSR(csr *certificatesv1.CertificateSigningRequest) bool {
-	nsn := types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}
-	csrPEM, _ := pem.Decode(csr.Spec.Request)
-	if csrPEM == nil {
-		klog.Errorf("Failed to PEM-parse the CSR block in .spec.request: no CSRs were found in %s", nsn)
-		return false
+	commonNamePrefixes := []string{}
+	// initialize Sets from slices
+	for i, v := range csrAcceptanceConditions {
+		csrAcceptanceConditions[i].groupsSet = sets.New[string](v.Groups...)
```

Comment:
I don't think the `csrAcceptanceConditions` parameter should be updated here.
Set the `groupsSet` and `userPrefixesSet` in the same place csrAcceptanceConditions is created (maybe extract that to a separate function).